cli/azd/pkg/infra/provisioning/terraform/terraform_provider.go

```diff
@@ -170,6 +171,27 @@ func (t *TerraformProvider) Deploy(ctx context.Context, deployment *DeploymentPl
 		return nil, fmt.Errorf("reading backend config: %w", err)
 	}
 
+	// Create parameter overrides file if needed.
+	overrides := make(map[string]interface{}, len(deployment.Deployment.Parameters))
+	for k, v := range deployment.Deployment.Parameters {
+		if v.Override {
+			overrides[k] = v.Value
+		}
+	}
+	if len(overrides) > 0 {
+		if terraformDeploymentData.overrideParametersFilePath == "" {
+			tmp, err := os.CreateTemp("", "*.json")
+			if err != nil {
+				return nil, fmt.Errorf("creating parameter overrides file: %w", err)
+			}
+			terraformDeploymentData.overrideParametersFilePath = tmp.Name()
+			defer func() {
+				_ = tmp.Close()
+				_ = os.Remove(tmp.Name())
+			}()
+		}
```

Comment:
Curious to understand the choice to create a var file vs using the terraform `-var` flags?

Reply:
I was also wondering this - I suspect the reason is that we have to end up creating a vars file anyway (as part of our existing logic to do environment variable substitution on the checked in tf-vars file) but I do think it is a more straight froward model for folks to grok to skip all this logic and just say that these --parameter commands map directly to -var flags on `plan`.